ambari-server/src/main/java/org/apache/ambari/server/orm/entities/RepoDefinitionEntity.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ambari.server.orm.entities;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.persistence.TableGenerator;
+
+import org.apache.ambari.server.state.stack.RepoTag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Entity
+@Table(name = "repo_definition", uniqueConstraints = {
+})
+@TableGenerator(name = "repo_definition_id_generator",
+    table = "ambari_sequences",
+    pkColumnName = "sequence_name",
+    valueColumnName = "sequence_value",
+    pkColumnValue = "repo_definition_id_seq",
+    initialValue = 0
+)
+public class RepoDefinitionEntity {
+  private static final Logger LOG = LoggerFactory.getLogger(RepoDefinitionEntity.class);
+
+  @Id
+  @Column(name = "repo_definition_id", nullable = false)
+  @GeneratedValue(strategy = GenerationType.TABLE, generator = "repo_definition_id_generator")
+  private Long id;
+
+  @ManyToOne
+  @JoinColumn(name = "repo_os_id", referencedColumnName = "repo_os_id", nullable = false)
+  private RepoOsEntity repoOs;
+
+  @Column(name = "repo_name", nullable = false)
+  private String repoName;
+
+  @Column(name = "repo_id", nullable = false)
+  private String repoID;
+
+  @Column(name = "base_url", nullable = false)
+  private String baseUrl;
+
+  @Column(name = "mirrors")
+  private String mirrors;
+
+  @Column(name = "distribution")
+  private String distribution;
+
+  @Column(name = "components")
+  private String components;
+
+  @Column(name = "unique_repo", nullable = false)
+  private short unique = 0;
+
+  @OneToMany(cascade = CascadeType.ALL, mappedBy = "repoDefinitionEntity")
+  private List<RepoTagEntity> repoTagEntities = new ArrayList<>();
```

Comment:
This should not be a whole new entity.  You should be able to use a CollectionTable (see AlertTargetEntity alertStates for an example)